tests/integration/conftest.py

```diff
@@ -51,15 +54,37 @@ def pytest_collection_modifyitems(config, items):
 
 
 @pytest.fixture(scope="session", autouse=True)
-def logging_setup():
+def test_env_setup():
+    _logging_setup()
+    old_state = libnmstate.show()
+    _empty_net_state()
+    _ethx_init()
+    yield
+    libnmstate.apply(old_state, verify_change=False)
+    _diff_initial_state(old_state)
+
+
+def _empty_net_state():
+    """
+    Remove existing DNS, routes, route rules in case it interference tests.
+    """
+    desired_state = libnmstate.show()
+    desired_state[DNS.KEY] = {DNS.CONFIG: {}}
+    desired_state[Route.KEY] = {
+        Route.CONFIG: [{Route.STATE: Route.STATE_ABSENT}]
```

Comment:
This means that it will apply it on all existing routes? Interesting.

Reply:
No defined means match all.